pkgs/top-level/all-packages.nix

```diff
@@ -14345,27 +14345,25 @@ let
 
   VisualBoyAdvance = callPackage ../misc/emulators/VisualBoyAdvance { };
 
-  # Wine cannot be built in 64-bit; use a 32-bit build instead.
-  wineStable = callPackage_i686 ../misc/emulators/wine/stable.nix {
-    bison = bison2;
+  # Wine defaults to a mixed 64 / 32 build on x86_64 and to pure 32 on x86
+  wine = callPackage ../misc/emulators/wine {
+    wineRelease = config.wine.release or "stable";
+    wineBuild = config.wine.build or (if system == "x86_64-linux" then "wineWow" else "wine32");
   };
+  wineStable = wine.override { wineRelease = "stable"; };
```

Comment:
`wineStable` is fixed to the stable release branch, while the release branch of `wine` is subject to selection through `config.wine`